src/wp-admin/includes/class-wp-users-list-table.php

```diff
@@ -499,8 +499,10 @@ public function single_row( $user_object, $style = '', $role = '', $numposts = 0
 				);
 			}
 
-			// Add a link to send the user a reset password link by email.
-			if ( get_current_user_id() !== $user_object->ID
+			// Add a link to send the user a reset password link by email if the user is allowed to get his password reset.
+			$allow = apply_filters( 'allow_password_reset', true, $user_object->ID );
+			if ( true === $allow
```

Comment:
```suggestion
			/** This filter is documented in src/wp-includes/user.php */
			$allow_password_reset = apply_filters( 'allow_password_reset', true, $user_object->ID );

			// Add a link to send the user a reset password link by email if the user is allowed to get his password reset.
			if ( true === $allow_password_reset
```
- Added a note of where this filter is documented.
- Renamed the `$allow` variable to `$allow_password_reset` for readability.